unbxdSearch.js

```diff
@@ -1146,7 +1164,18 @@ var unbxdSearchInit = function(jQuery, Handlebars){
       if(doPush){
 	var finalquery = this.options.noEncoding ? urlobj.query : this.encode( urlobj.query );
 	if(this.isHistory){
-	  history.pushState(this.params,null,location.protocol + "//" + location.host + location.pathname + "?" + finalquery);
+	  if( self.options.retainbaseParam == true ){
+	    var finalBaseParams = '';
+	    for( i in urlobj.baseParams){
+	      if(urlobj.baseParams.hasOwnProperty(i)){
+	        finalBaseParams += "&" + i + "=" + urlobj.baseParams[i];
+	      }
+	    }
+	    finalquery += finalquery + this.options.noEncoding ? finalBaseParams : this.encode(finalBaseParams);
+	    history.pushState(this.params, null, location.protocol + "//" + location.host + location.pathname + "?" + finalquery);
+	  }else{
+	    history.pushState(this.params,null,location.protocol + "//" + location.host + location.pathname + "?" + finalquery);
+	  }
```

Comment:
code wise, only issue I see is, the above lines, 1177 & 1175 are same. So, end the if loop before 'history.pushState()' and keep it outside of loop. Should work fine, and no duplication of code. Other than that, mostly good for me.